packages/tpetra/core/src/Tpetra_RowMatrix_def.hpp

```diff
@@ -222,6 +222,12 @@ namespace Tpetra {
         C = rcp (new crs_matrix_type (C_rowMap, 0, DynamicProfile,
                                       constructorSublist));
       }
+#else
+      TEUCHOS_TEST_FOR_EXCEPTION(!A_rowMap->isSameAs (*B_rowMap),
```

Comment:
We've already tested this condition above.  Just do this:
```c++
TEUCHOS_TEST_FOR_EXCEPTION(true, std::invalid_argument, "error message goes here");
```

Reply:
Fair point.